File: backend/src/main/java/com/resourcex/resourcex/service/DisputeService.java
package com.resourcex.resourcex.service;

import com.resourcex.resourcex.dto.request.CreateDisputeRequest;
import com.resourcex.resourcex.dto.request.ResolutionRequest;
import com.resourcex.resourcex.dto.response.DisputeResponse;
import com.resourcex.resourcex.entity.Dispute.DisputeStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface DisputeService {

    DisputeResponse createDispute(CreateDisputeRequest request);

    DisputeResponse getDisputeById(Long disputeId);

    Page<DisputeResponse> getAllDisputes(Pageable pageable);

    List<DisputeResponse> getDisputesByUser(Long userId);

    List<DisputeResponse> getDisputesByStatus(DisputeStatus status);

    DisputeResponse resolveDispute(Long disputeId, ResolutionRequest request);

    DisputeResponse closeDispute(Long disputeId);

    void deleteDispute(Long disputeId);

    void followUpStaleDisputes(java.time.LocalDateTime threshold);
}

File: frontend/web/lib/services/disputeService.ts
import api from "../api";
import { DisputeResponse } from "../../types/dispute";

export const disputeService = {
  getAllDisputes: async (): Promise<DisputeResponse[]> => {
    const response = await api.get<DisputeResponse[]>("/disputes");
    return response.data;
  },

  getDisputeById: async (disputeId: number): Promise<DisputeResponse> => {
    const response = await api.get<DisputeResponse>(`/disputes/${disputeId}`);
    return response.data;
  },

  createDispute: async (disputeData: {
    bookingId: number;
    reason: string;
  }): Promise<DisputeResponse> => {
    const response = await api.post<DisputeResponse>("/disputes", disputeData);
    return response.data;
  },

  resolveDispute: async (
    disputeId: number,
    resolutionData: {
      resolution: string;
      resolvedByUserId: number;
    }
  ): Promise<DisputeResponse> => {
    const response = await api.post<DisputeResponse>(
      `/disputes/${disputeId}/resolve`,
      null,
      {
        params: {
          resolution: resolutionData.resolution,
          userId: resolutionData.resolvedByUserId,
        },
      }
    );
    return response.data;
  },
};
